### PR TITLE
Use `Mapping` in ERC-721 Example

### PR DESCRIPTION
**Pull request by HCastano**

### CODE REVIEW

File: examples/erc721/lib.rs [hunk 1]
```diff
@@ -130,12 +129,18 @@ mod erc721 {
...
-            }
+            ink_lang::codegen::initialize_contract(|_| {
+                Self {
+                    // TODO: How should this be properly initialized?
+                    //
+                    // Turns out that `token_owner` and `token_approvals` have the same type
+                    // signature and they're both Default initialized to ROOT_KEY [0u8; 32]. This
+                    // means that they end up operating on the same storage keys :(
+                    // token_approvals: Default::default(),
+                    token_approvals: Mapping::new([1u8; 32].into()),
```
Comment (HCastano) [inline examples/erc721/lib.rs:140]: @Robbepop looks like we need another piece of "entropy" when [assigning storage keys](https://github.com/paritytech/ink/blob/78b45330e453500a759a2b53b7ab6f16bfb91caa/crates/storage/src/lazy/mapping.rs#L115) in the `Mapping`. 

I haven't been able to figure out what we could use though, do you have any ideas?

File: examples/erc721/lib.rs [hunk 2]
```diff
@@ -439,8 +449,10 @@ mod erc721 {
                     .expect("Cannot get accounts");
             // Create a new contract instance.
             let mut erc721 = Erc721::new();
+            dbg!(erc721.get_approved(1));
```
Comment (HCastano) [inline examples/erc721/lib.rs:452]: I'll get rid of these once the initialize thing is sorted

File: examples/erc721/lib.rs (hunk 1 above)
Comment (HCastano) [reply examples/erc721/lib.rs]: I noticed this "bug" when before swapping the initialization routine to actually use `ink_lang::codegen::initialize_contract`. By using it the `Mapping`'s get intialized with different `offset_key`s, so false alarm 😅

File: examples/erc721/lib.rs [hunk 3]
```diff
@@ -297,16 +310,26 @@ mod erc721 {
...
-            let entry = owned_tokens_count.entry(*to);
-            increase_counter_of(entry);
-            vacant_token_owner.insert(*to);
+
+            let count = owned_tokens_count
+                .get(to)
+                .map(|mut c| {
+                    c += 1;
+                    c
+                })
```
Comment (athei) [inline examples/erc721/lib.rs:327]:
```suggestion
                .map(|c| c + 1)
```

File: examples/erc721/lib.rs [hunk 4]
```diff
@@ -325,19 +348,14 @@ mod erc721 {
...
-                Ok(())
+
+            if approved {
+                self.operator_approvals.insert((&caller, &to), &approved);
             } else {
-                match self.operator_approvals.insert((caller, to), approved) {
-                    Some(_) => Err(Error::CannotInsert),
-                    None => Ok(()),
-                }
+                self.operator_approvals.remove((&caller, &to));
```
Comment (athei) [inline examples/erc721/lib.rs:355]: Do we still need the `bool` as the value in the mapping after this change? Couldn't it be just `()` now that we only check for existence?

File: examples/erc721/lib.rs [hunk 5]
```diff
@@ -349,43 +367,40 @@ mod erc721 {
...
                 from: caller,
                 to: *to,
                 id,
             });
+
             Ok(())
         }
 
         /// Removes existing approval from token `id`.
         fn clear_approval(&mut self, id: TokenId) -> Result<(), Error> {
```
Comment (athei) [inline examples/erc721/lib.rs:391]: It no longer can return an error.

**Review by athei — APPROVED** [review]

**Pull request merged**